Issue 2606: Added UTF-8 string serializer.

claudiofahey opened the pull request:
Signed-off-by: Claudio Fahey <claudio.fahey@emc.com>

**Change log description**
Added UTF-8 string serializer.

**Purpose of the change**
Currently, events that consist of strings are serialized in Pravega using JavaSerializer. This is not ideal because it is difficult to use in non-Java environments and adds unnecessary framing. This change adds a new UTF-8 string serializer.

**What the code does**
This change adds a new UTF-8 string serializer.

**How to verify it**
See UTF8StringSerializerTest.

tkaitchuck commented on client/src/main/java/io/pravega/client/stream/impl/JavaSerializer.java:
```diff
@@ -22,6 +22,8 @@
 
 /**
  * An implementation of {@link Serializer} that uses Java serialization.
+ * For strings, {@link UTF8StringSerializer} is preferred as it is works
```
I don't think this comment is needed.

tkaitchuck commented on client/src/main/java/io/pravega/client/stream/impl/UTF8StringSerializer.java:
```diff
@@ -0,0 +1,33 @@
+
+    @Override
+    public String deserialize(ByteBuffer serializedValue) {
+        return new String(serializedValue.array(), StandardCharsets.UTF_8);
```
This is incorrect. A byte buffer may be backed by an array but represents a subset of it. You need to get the correct offset by adding arrayOffset() + position() and the length as bytesRemaining()

tkaitchuck replied:
Please also add a test for this case

claudiofahey commented:
I have corrected the ByteBuffer access using the technique from https://stackoverflow.com/a/17354950/5890553.
I also added the test testByteArrayWithOffset.
